src/promptflow/promptflow/executor/_line_execution_process_pool.py

```diff
@@ -208,65 +177,116 @@ def __exit__(self, exc_type, exc_val, exc_tb):
             self._pool.close()
             self._pool.join()
 
-    def _timeout_process_wrapper(self, run_start_time: datetime, task_queue: Queue, timeout_time, result_list):
-        healthy_ensured_process = HealthyEnsuredProcess(self._executor_creation_func, self.context)
-        healthy_ensured_process.start_new(task_queue)
+    def _get_process_info(self, index):
+        while True:
+            try:
+                process_id = self._process_info[index].process_id
+                process_name = self._process_info[index].process_name
+                process_info = ProcessInfo(index=index, process_id=process_id, process_name=process_name)
+                return (process_info.index, process_info.process_id, process_info.process_name)
+            except KeyError:
+                continue
+            except Exception as e:
+                bulk_logger.warning(f"Unexpected error occurred while get process info. Exception: {e}")
+
+    def handle_line_timeout(self, line_number, timeout_time, inputs, run_id, start_time, result_list):
+        bulk_logger.warning(f"Line {line_number} timeout after {timeout_time} seconds.")
+        ex = LineExecutionTimeoutError(line_number, timeout_time)
+        result = self._generate_line_result_for_exception(inputs, run_id, line_number, self._flow_id, start_time, ex)
+        result_list.append(result)
+
+    def handle_process_crashed(self, line_number, inputs, run_id, start_time, result_list):
+        bulk_logger.warning(f"Process crashed while executing line {line_number},")
+        ex = ProcessCrashError(line_number)
+        result = self._generate_line_result_for_exception(inputs, run_id, line_number, self._flow_id, start_time, ex)
+        result_list.append(result)
+
+    def _monitor_process_alive(self, process_id):
+        if not psutil.pid_exists(process_id):
```

Comment:
This function can be simplified
```suggestion
    return not psutil.pid_exists(process_id)
```

Reply:
fixed